orchestra/workflow/load.py

```diff
@@ -0,0 +1,161 @@
+        )
+        if required_certs.count() != len(required_cert_slugs):
+            raise WorkflowError(
+                'Certification {} requires non-existent certification.'
+                .format(certification_data['slug']))
+        cert.required_certifications = list(required_certs)
+
+    # Load the desired versions
+    seen_versions = set()
+    requested_versions = set(versions)
+    for version_data in data['versions']:
+        version_slug = version_data['slug']
+        seen_versions.add(version_slug)
+        if versions is None or version_slug in requested_versions:
+            load_workflow_version(version_data, workflow, force=force)
+    missing_versions = requested_versions - seen_versions
+    if versions is not None and missing_versions:
+        raise WorkflowError('Invalid versions requested: {}'
+                            .format(missing_versions))
+
+
+def load_workflow_version(version_data, workflow, force=False):
+    # Create the version object
+    version, version_created = create_or_update(
+        WorkflowVersion,
+        {
+            'slug': version_data['slug'],
+            'workflow': workflow
+        },
+        {
+            'name': version_data['name'],
+            'description': version_data['description']
+        }
+    )
+
+    if not version_created:
+        if not force:
+            raise WorkflowError('Version {} already exists'
+                                .format(version_data['slug']))
+
+        # Check that the versions are safe to merge
+        new_step_slugs = set(s['slug'] for s in version_data['steps'])
+        old_step_slugs = set(
+            WorkflowStep.objects
+            .filter(workflow_version=version)
+            .values_list('slug', flat=True)
+        )
+        if new_step_slugs != old_step_slugs:
+            raise WorkflowError('Even with --force, cannot change the steps '
+                                'of a workflow. Drop and recreate the '
+                                'database to reset, or create a new version '
+                                'for your workflow.')
+
+    # Create or update the version steps.
+    for step_data in version_data['steps']:
+        step, step_created = create_or_update(
+            WorkflowStep,
+            {
+                'slug': step_data['slug'],
+                'workflow_version': version
+            },
+            {
+                'name': step_data['name'],
+                'description': step_data['description'],
+                'is_machine': step_data.get('is_machine', False),
+                'execution_function': step_data.get('execution_function', {}),
+                'assignment_policy': step_data.get('assignment_policy', {}),
+                'review_policy': step_data.get('review_policy', {}),
+                'user_interface': step_data.get('user_interface', {}),
+            }
+        )
+
+        # TODO(dhaas): should we prevent updates to these?
```

Comment:
i could imagine over the course of developing a project, certification dependencies could be added or removed, and we don't want to uncertify workers automatically just because of that.

also: i'd be OK with us completely removing required_certifications, because we're just not there yet (TM) in terms of having dependency trees.  that said, i'm also OK just leaving things and not trying to be too prescriptive just yet